### ISSUE TITLE
posting method to /execute is broken or removed in Appium 1.5.0-beta8 ?

### ISSUE THREAD

**Issue by KazuCocoa** [appium/appium#5920]
`execute` post like `/wd/hub/session/a7e3d797-9ea2-4383-83dd-6ff227dbd615/execute` is broken or removed in Appium 1.5.0-beta8 ?

The following log of UIAutomation post works well in Appium 1.4.16, but Appium 1.5.0-beta8 doesn't.

```
info: [debug] Responding to client with success: {"status":0,"value":true,"sessionId":"a7e3d797-9ea2-4383-83dd-6ff227dbd615"}
info: <-- POST /wd/hub/session/a7e3d797-9ea2-4383-83dd-6ff227dbd615/execute 200 963.667 ms - 76 {"status":0,"value":true,"sessionId":"a7e3d797-9ea2-4383-83dd-6ff227dbd615"}
info: --> POST /wd/hub/session/a7e3d797-9ea2-4383-83dd-6ff227dbd615/execute {"script":"$.mainApp().alert().buttons().length > 0","args":[]}
```

When I set debug log-level and compared the logs between Appium 1.4.16 and Appium 1.5.0-beta8. Then I found `execute` post didn't called in Appium 1.5.0-beta8 at all.

Is this a bug or new spec ?
If this is new spec, do you have any diffs between Appium 1.4.x to 1.5.0 ?
If yes, the diffs will help me checking myself.

# my env
- Appium 1.4.16 or Appium 1.5.0-beta8
- client: ruby_lib 8.0.0
- Xcode6.4, Mac OS 10.10(yosemite)

**Comment by scottdixon** [comment]
Can you please post your appium logs?

I just ran execute with success on Xcode 7 / iOS9 / Appium 1.5 beta 9.

Also, you can see the 1.5 changelog WIP here: https://github.com/appium/appium/blob/1.5/CHANGELOG.md

**Comment by KazuCocoa** [comment]
I will re-run with Appium 1.5 beta 9 and share the log with Gist.
And thanks for changelog !

**Comment by KazuCocoa** [comment]
@scottdixon I tried again and I attach gist to logs. Could you check them ?

---

with Appium 1.5.0 beta9: https://gist.github.com/KazuCocoa/17e059277295937151cd
with Appium 1.4.16: https://gist.github.com/KazuCocoa/e7da9a77d6a715c6a172

------

- With Appium 1.5.0 beta9, failed test
    1. Launch iOS simulator 8.4 <= first time failed, but second time succeeded.
    2. Try to tap yystem alert dialog <= failed

- With Appium 1.4.16, succeeded test
    1. Launch iOS simulator 8.4
    2. Try to tap yystem alert dialog
    3. Run test cases

- env
    - Appium 1.4.16 or Appium 1.5.0-beta8
    - client: ruby_lib 8.0.0
    - Xcode6.4, Mac OS 10.10(yosemite)

------

Xcode7 also failed test as Xcode6.4.
`implicit_wait` looks failed on Appium 1.5.0 beta9...

**Comment by KazuCocoa** [comment]
ping @scottdixon

**Comment by scottdixon** [comment]
Regarding launch retries, try increasing the timeout > 90 seconds.

I can't see an attempting to tap system alert - can you share your test code?

**Comment by KazuCocoa** [comment]
The system alert means confirmation of push notification. The user should select "Don't Allow" or "OK".
In my case, the alert is displayed after installing and launching the app. And after tapping "don't allow" or "ok", some test scenarios run.

A scenario of "after tapping allow or ok," works the following steps.

1. install and launch the app via Appium
2. `wait_true` the alert is displayed with `execute_script`
3. if the alert is shown, we call method `alert_accept`

`wait_true`, `execute_script` and `alert_accept` is provided by `ruby_lib`.

In Appium 1.4.16, the scenario works well. But Appium 1.5.0-beta9 doesn't. The difference of them is called `/execute` or not.

https://gist.github.com/KazuCocoa/e7da9a77d6a715c6a172#file-20151224_logs_appim_1-4-16-txt-L139


Here is a simple test code.

```ruby
Appium::Driver.new(caps).start_driver # set some caps regarding with iOS
Appium.promote_appium_methods AppiumCustomDriver # extend ruby_lib method to my AppiumCustomDriver

# start scenario

# failed `execute_script`
wait_true(timeout: 90, interval: 0.5,
    message: "error_message") { execute_script("$.mainApp().alert().buttons().length > 0") }
alert_accept

# continue to some test scenarios....
find_element(accessibility_id: "my_id").click # this method works
...
```

without `wait_true` , them iOS simulator will crash.

**Comment by KazuCocoa** [comment]
Reproduced yet with Appium 1.5.0-beta11.

This scenario is same above one.

https://gist.github.com/KazuCocoa/475baba8eac8484197e5#file-20160114_logs_appium_1-5-0-beta11-txt-L56
According to attempted log, Appium server try to take screenshot as teardown test scenarios.

**Comment by scottdixon** [comment]
It looks like the test is failing and tearing down before execute is called. Appium log isn't reporting anything obvious. Can you share your entire test code?

**Comment by KazuCocoa** [comment]
Yes, i can.
I prepared simple test case reproduce this issue. And I add sample iOS(Swift) application for this issue. Please check the following repository.

- https://github.com/KazuCocoa/scenarioForAppiumIssue5920
- iOS app: https://github.com/KazuCocoa/simplePushAppForIssue
    - built app is included in `scenarioForAppiumIssue5920`

**Comment by KazuCocoa** [comment]
Does this issue proceed ?

**Comment by scottdixon** [comment]
Thanks for all the code. I'll be investigating further this week.

**Comment by KazuCocoa** [comment]
thanks :smile:

**Comment by scottdixon** [comment]
Thanks for all the code @KazuCocoa. I was able to repro the issue - appium-ios-driver wasn't returning the correct caps so the ruby client thought JS was disabled.

I've published the fix in `appium-ios-driver@1.8.2`

It will be available in the next Appium release. Alternative you can use appium-ios-driver or appium from source.

Let me know if any issues.

**Comment by KazuCocoa** [comment]
Hm... I understood. Thanks so much !
I will share issues with any samples if I find any issues with new version of Appium :)
